audioswitch/src/main/java/com/twilio/audioswitch/AudioSwitch.kt

```diff
@@ -134,7 +147,24 @@ class AudioSwitch {
         this.audioDeviceManager = audioDeviceManager
         this.wiredHeadsetReceiver = wiredHeadsetReceiver
         this.bluetoothHeadsetManager = headsetManager
+        this.preferredDeviceList = getPreferredDeviceList(preferredDeviceList)
         logger.d(TAG, "AudioSwitch($VERSION)")
+        logger.d(TAG, "Preferred device list = ${this.preferredDeviceList.map { it.simpleName }}")
+    }
+
+    private fun getPreferredDeviceList(preferredDeviceList: List<Class<out AudioDevice>>):
+            List<Class<out AudioDevice>> {
+        require(preferredDeviceList.isNotEmpty() && hasNoDuplicates(preferredDeviceList))
```

Comment:
I think empty is okay. Providing an empty list results in the default preferred list being used.

Reply:
Let's also ensure that an `IllegalArgumentException` is thrown when the list has duplicates.

Reply:
Yep, I think that makes sense for an empty list. And an `IllegalArgumentException` does get thrown in the case of a duplicate.